### ISSUE TITLE
Duplicate analysis bundles for 36 primary bundles in 03/01 scale test

### ISSUE THREAD

**Issue by parthshahva** [HumanCellAtlas/ingest-central#339]
Issue: Some of the analysis envelopes in the scale test seem to have exported multiple bundles even though only one workflow ran.
Submission ID: 5c78bb0206bd55000b84d84e
Project UUID: 4c21b606-4a30-46a6-80a1-5cb1074badf9
Environment: Integration
Test identifier: scale/ss2_2000/2019-03-01T04:54:21Z

There are 36 primary bundles with 2 results bundles from this scale test. 5 examples:

Analysis Envelope with 2 bundles exported: 5c7968e906bd55000b851014
primary: 22483e84-223c-4d18-bbe3-faf19d3b65a4.2019-03-01T163942.194011Z secondary: ['6157d760-dc96-499b-87ea-7b47cc0def35.2019-03-01T203153.264877Z', '0ca0e5be-c312-4825-b96f-c881a5cf342f.2019-03-01T203153.648593Z']

Analysis Envelope with 2 bundles exported: 5c796a8006bd55000b8517c8
primary: a7ea19d1-eb88-4006-90d2-210551e0cc7e.2019-03-01T164556.945204Z secondary: ['f99a7f8d-f1a5-41ee-99d0-1483fa4f547f.2019-03-01T213627.893804Z', 'd13c3747-b249-450a-b7b0-c949617f61ae.2019-03-01T213627.997172Z']

Analysis Envelope with 2 bundles exported: 5c796ba306bd55000b851c5e
primary: bcb46509-60ec-4e0d-9969-e94daeb6f2bc.2019-03-01T164606.261093Z secondary: ['db0cdb21-0bb4-4c74-b439-78d34e396d77.2019-03-01T214907.384667Z', '61d05a41-163e-42e1-8c9c-5d4265ca8212.2019-03-01T214907.786165Z']

Analysis Envelope with 2 bundles exported: 5c796ac906bd55000b85193a
primary: dd9d4ab2-b9de-474d-803d-c75ec7fd2e9f.2019-03-01T164758.846047Z secondary: ['e30dbb46-1d59-48db-aac8-b4b75ff73e2e.2019-03-01T214009.432221Z', '1a3dd201-b2e8-48a9-994b-e46dbfd5f9d8.2019-03-01T214009.543756Z']

Analysis Envelope with 2 bundles exported: 5c796ab506bd55000b851842
primary: e19d1061-5a37-4a01-8fd9-d792ce35f3f2.2019-03-01T164857.697033Z secondary: ['e2fbb37e-9c99-44d3-b23a-cde0c92e6f47.2019-03-01T212845.003767Z', '1c018723-fd75-434c-a152-85dfb09bc8f4.2019-03-01T212843.652387Z']

**Comment by aaclan-ebi** [comment]
It seems that the exporter receives the same analysis message twice. It's possible that:
- Ingest Core is generating 2 messages for the same analysis
- Envelope is being submitted twice (which triggers the finding of analysis and generation of exporter messages)

Unfortunately, I couldn't confirm thru our logs but I've enabled INFO logging in Ingest Core to verify what is happening. @parthshahva could we just note of this issue for now and see if this is going to happen again.

**Comment by parthshahva** [comment]
Sounds good. We can see if this happens again this week

**Comment by aaclan-ebi** [comment]
Update: Looks like we're logging the index and total analysis bundles in a submission in the Exporter and I can confirm that 1 is not happening. Looks like it's 2. But I'm still uncertain why. We have a check that a submission could no longer be set to `Submitted` if its state is already `Submitted`. Still need to take a closer look...

**Comment by aaclan-ebi** [comment]
Confirmed from the state tracker logs that submission envelopes are being submitted only once.

Looks like there is really a possibility of message duplication in RabbitMQ https://www.rabbitmq.com/reliability.html It’s suggesting to handle messages in an idempotent way. I wonder why it’s only now that we’re observing this. Anyway, I created a separate ticket to fix the issue. Might not be fixed 'til next scale test schedule.

https://app.zenhub.com/workspaces/dcp-backlogs-5ac7bcf9465cb172b77760d9/issues/humancellatlas/ingest-central/351
